include/openthread/ip6.h

```diff
@@ -237,6 +237,7 @@ typedef struct otMessageInfo
     bool    mIsHostInterface : 1;   ///< TRUE if packets sent/received via host interface, FALSE otherwise.
     bool    mAllowZeroHopLimit : 1; ///< TRUE to allow IPv6 Hop Limit 0 in `mHopLimit`, FALSE otherwise.
     bool    mMulticastLoop : 1;     ///< TRUE to allow looping back multicast, FALSE otherwise.
+    uint8_t mDscp : 6;              ///< Diff Services Code Point in IPv6 header (provided for a rx msg, ignored on tx).
```

Comment:
Why does it say "ignored on tx" ? Shouldn't any DSCP value other than 0 be included in the 6lowpan header, so that it would be used when prepping the frame for Tx?

Reply:
That's a different topic. I don't want to change behavior on tx.

Specially since there may be app layer code using the previous format of `otMessageInfo` and not set the new field `mDscp` when it prepares an instance of this `struct` and pass it in OT API to send a `Message`. Developers hopefully `memset` the `otMessageInfo` to zero but not guranteed.

To be honest, I was reluctant to add this in `otMessageInfo` which is a OT public definition, but could not find any better way. I will think more on this to see I can find another way, I prefer not to change the public definition.

Reply:
Ok, I didn't know the background of this. If it is potentially uninitialized data then it makes sense to ignore.

So if I'm correct now a public user of the OT API cannot provide a DSCP value for outgoing application related IPv6 packets, while the stack internally can use DSCP. (At the moment only applied for TMF messages.)
And the public user of the OT API can read out the DSCP value used for incoming application related IPv6 messages? (E.g. for traffic coming from an internet host)

Reply:
OpenThread defines its own priority enum, stores that in the message object, and maps to DSCP values internally. Can we reuse that?

Reply:
Actually looking more into this, I don't think we need this PR. 

We already determine the priority in `MeshForwarder` for TMF messages (from DSCP) and set it on the `Message` in:
https://github.com/openthread/openthread/blob/f0232d0f79f28042727455bfdaf92563b6fe6595/src/core/thread/mesh_forwarder.cpp#L1625-L1637
This was already added in https://github.com/openthread/openthread/pull/7869 :)